### PR TITLE
Add more things

### PR DESCRIPTION
**Pull request by william-liebenberg**
Interfaced a bunch of things
Added ILogger implementations
Code cleanup

### CODE REVIEW

**Review by william-liebenberg — COMMENTED** [review]
Ahh so many places to add comments...

**Comment by vludax** [comment]
yo :D

**Review by vludax — APPROVED** [review]

**Comment by vludax** [comment]
lol so any random can approve the changes

**Comment by william-liebenberg** [comment]
Ah you can set up who can/can't merge the pull requests
the layout is a bit crazy... still digging through stuff

File: GuruFX/GuruFX.Core/Entity.cs [hunk 1]
```diff
@@ -3,23 +3,69 @@
...
+				return false;
+			}
+
+			// add the component
+			Components.Add(component.InstanceID, component);
+
+			return true;
+		}
+
+		public bool AddComponents<T>(T[] components)
```
Comment (william-liebenberg) [inline GuruFX/GuruFX.Core/Entity.cs:60]: @vludax whoooo mentions work as well :)

**Review by william-liebenberg — COMMENTED** [review]
Another comment

File: GuruFX/GuruFX.Core/Entity.cs (hunk 1 above)
Comment (william-liebenberg) [reply GuruFX/GuruFX.Core/Entity.cs:60]: maybe im old... this code review interface is taking a bit of getting used to

Comment (vludax) [reply GuruFX/GuruFX.Core/Entity.cs:60]: I think it's just different -- wouldn't take a week of active usage to become familiar with it ;)

**Pull request merged**